apps/client/src/app/pages/portfolio/activities/import-activities-dialog/import-activities-dialog.component.ts

```diff
@@ -39,7 +43,39 @@ export class ImportActivitiesDialog implements OnDestroy {
     this.dialogRef.close();
   }
 
-  public onImport() {
+  public async onImportActivities() {
+    try {
+      await this.importActivitiesService.importSelectedActivities(
+        this.selectedActivities
+      );
+
+      this.snackBar.open(
+        '✅ ' + $localize`Import has been completed`,
+        undefined,
+        {
+          duration: 3000
+        }
+      );
+    } catch (error) {
+      this.snackBar.open(
+        $localize`Oops! Something went wrong.` +
+          ' ' +
+          $localize`Please try again later.`,
+        $localize`Okay`,
+        { duration: 3000 }
+      );
+    } finally {
+      this.dialogRef.close();
+    }
+  }
+
+  public onReset() {
+    this.details = [];
+    this.errorMessages = [];
+    this.isFileSelected = false;
+  }
+
+  public onSelectFile() {
```

Comment:
Change the snack bar message below from 
```
'⏳ ' + $localize`Importing data...`
```
to
```
'⏳ ' + $localize`Validating data...`
```

Reply:
Done